plugin/src/main/cpp/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,252 @@
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+uint64_t OpenXRFbSpaceWarpExtensionWrapper::_set_system_properties_and_get_next_pointer(void *p_next_pointer) {
+	system_space_warp_properties.next = p_next_pointer;
+	return reinterpret_cast<uint64_t>(&system_space_warp_properties);
+}
+
+uint64_t OpenXRFbSpaceWarpExtensionWrapper::_set_projection_views_and_get_next_pointer(int p_view_index, void *p_next_pointer) {
+	if (fb_space_warp_ext && enabled) {
+		space_warp_info[p_view_index].next = p_next_pointer;
+		return reinterpret_cast<uint64_t>(&space_warp_info[p_view_index]);
+	} else {
+		return reinterpret_cast<uint64_t>(p_next_pointer);
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_session_created(uint64_t p_instance) {
+	if (!fb_space_warp_ext) {
+		return;
+	}
+
+	String graphics_api_name = get_openxr_api()->get_graphics_api_name();
+	if (graphics_api_name != "OpenGL") {
+		UtilityFunctions::print("Disabling XR_FB_space_warp extension; this extension is only implmented for OpenGL");
+		fb_space_warp_ext = false;
+		return;
+	}
+
+	get_openxr_api()->register_projection_views_extension(this);
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_session_destroyed() {
+	if (fb_space_warp_ext) {
+		get_openxr_api()->unregister_projection_views_extension(this);
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_state_ready() {
+	ProjectSettings *project_settings = ProjectSettings::get_singleton();
+	bool is_project_setting_enabled = (bool)project_settings->get_setting_with_override("xr/openxr/extensions/application_space_warp");
+	if (!is_project_setting_enabled) {
+		fb_space_warp_ext = false;
+	} else {
+		enabled = true;
+	}
+
+	if (!fb_space_warp_ext || !enabled) {
+		return;
+	}
+
+	Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+	int view_count = openxr_interface->get_view_count();
+	int width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+	int height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+
+	String graphics_api_name = get_openxr_api()->get_graphics_api_name();
+	if (graphics_api_name == "OpenGL") {
+		motion_vector_swapchain_info = get_openxr_api()->openxr_swapchain_create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_COLOR_ATTACHMENT_BIT, GL_RGBA16F, width, height, 1, view_count);
+		motion_vector_depth_swapchain_info = get_openxr_api()->openxr_swapchain_create(0, XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT, GL_DEPTH24_STENCIL8, width, height, 1, view_count);
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_main_swapchains_created() {
+	if (!fb_space_warp_ext) {
+		return;
+	}
+
+	Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+	int view_count = openxr_interface->get_view_count();
+
+	space_warp_info = memnew_arr(XrCompositionLayerSpaceWarpInfoFB, 2);
```

Comment:
I don't see where this array is `memdelete()`d?

Reply:
Whoops, added the deletion to `_on_session_destroyed()`!